spec/monitoring/metrics_spec.rb

```diff
@@ -1,15 +1,77 @@
 require 'rack/test'
 require 'prometheus/client/formats/text'
 require 'monitoring/prometheus'
+require 'monitoring/pub_sub'
+
+class SampleMetric
+  def setup(registry, pubsub)
+    registry.register(::Prometheus::Client::Gauge.new(
+      :test_gauge,
+      docstring: '...',
+      labels: [:test_label]
+    ))
+
+    pubsub.subscribe("sample_test_gauge") do |payload|
+      metric = registry.get(:test_gauge)
+      metric.set(payload[:value], labels: payload[:labels])
+    end
+  end
```

Comment:
I found myself wondering if we're doing too much here. I imagined it might be enough for the #setup unit tests to check if this method is called with the expected registry and pubsub.

I think I might have written in the issue or the design doc about the idea of being able to validate that pubsub events can result in changes on the registry. So I can see the value of those comprehensive tests on actual metric definitions (instead of on the test metric). My thoughts are we can keep this but strip it down to the bare minimum once we have an actual metric